### PR TITLE
fix(dlm):database not found

### LINKED ISSUE #1657: [Bug]: Data cleaning, view task details database field is empty
### ODC version

ODC424

### OB version

independent

### What happened?

Data cleaning, view task details database field is empty
image
image

### What did you expect to happen?

normally

### How can we reproduce it (as minimally and precisely as possible)?

chat privately

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
Simplified scheduled task return entities.
#### Which issue(s) this PR fixes:

Fixes #1657

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/factory/ScheduleResponseMapperFactory.java [hunk 1]
```diff
@@ -75,9 +75,9 @@ public ScheduleResponseMapper generate(@NonNull Collection<ScheduleEntity> entit
         List<UserEntity> userEntities = userRepository.findByUserIds(creators);
         Map<Long, UserEntity> userEntityMap = userEntities.stream().collect(
                 Collectors.toMap(UserEntity::getId, userEntity -> userEntity));
-        Set<Long> connectionIds = entities.stream().map(ScheduleEntity::getConnectionId).collect(Collectors.toSet());
-        Map<Long, ConnectionConfig> id2Datasource = connectionService.innerListByIds(connectionIds).stream().collect(
-                Collectors.toMap(ConnectionConfig::getId, o -> o));
+        Set<Long> databaseIds = entities.stream().map(ScheduleEntity::getDatabaseId).collect(Collectors.toSet());
+        Map<Long, Database> id2Database = databaseService.listDatabasesByIds(databaseIds).stream().collect(
```
[the remark this answers is not in the document]
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/factory/ScheduleResponseMapperFactory.java:79]: The method input parameters have been verified as non-null. At the same time, databaseId is a non-null field, so the collection cannot be empty.

**Review by krihy — COMMENTED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/factory/ScheduleResponseMapperFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/factory/ScheduleResponseMapperFactory.java:79]: non-null does not means not emtpy, only null check?

[the remark this answers is not in the document]
Comment (guowl3) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/factory/ScheduleResponseMapperFactory.java:82]: The caller has checked whether the collection is empty, and I also add a check in this method.

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**